src/python/geoclaw/surge/storm.py

```diff
@@ -548,6 +549,115 @@ def read_ibtracs(self, path, storm_name, year):
             self.storm_radius = numpy.where(ds.usa_roci >=0,
                 units.convert(ds.usa_roci.values,'nmi','m'),-1)
 
+    def read_emanuel(self, path, storm_name, verbose=False):
+        r"""Read in Kerry Emanuel's storm file
+
+        This reads in the netcdf-formatted tracks from Kerry Emanuel.
+        Correspondence September 2018. Original storms have been converted
+        from matlab to netcdf format by mlimb@rhg.com and are at
+        /gcs/../coastal/storm_tracks/emmanuel_projections_2018-09-12/netcdf_added_velocity/.
+
+        :Input:
+        - *path* (string) Path to the file to be read.
+        - *storm_name* (string) storm name containing information about
+        model, scenario, time period, and storm index within the file.
+            ex. ccsm4_rcp85_2007_2025_0, hadgem5_rcp45_2035_2045_10
+            model_scenario_period is equivalent to filename.
+
+        Assume these attributes exist in Emanuel dataset:
+           datetime
+           longstore
+           latstore
+           rmstrore
+           pstore (central pressure)
+           v_total_max_ms
+           bas (attribute)
+
+        :Raises:
+        - ImportError if xarray not found
+        - KeyError if provided storm name is not found in
+        given path (assumes path is correct)
+        """
+        # try/except copied from read_ibtracs
+        # imports that you don't need for other read functions
+        try:
+            import xarray as xr
+        except ImportError as e:
+            print("Emanuel tracks currently require xarray "
+                  "to work.")
+            raise e
+
+        def _convert_to_python_datetime(np_date):
+            '''Convert numpy.datetime64 to python datetime.datetime'''
+            (year, month, day_hour) = str(np_date.values).split('-')
+            day = day_hour[0:2]
+            hour = day_hour[3:5]
+            return datetime.datetime(int(year), int(month), int(day), int(hour))
+
+        storm_index = int(storm_name.split('_')[-1])
+
+        with xr.open_dataset(path, drop_variables=['time']) as ds:
+            try:
+                storm = (ds.sel(storm=storm_index)
+                     .drop('storm')
+                     .dropna(dim='time'))
+            except KeyError as e:
+                print("Provided storm name/index not found in "
+                      "the file.")
+                raise e
+
+            # set time
+            # convert from numpy to python datetime
+            # self.t is a list, as opposed to numpy array
+            self.t = [_convert_to_python_datetime(d) for d in storm['datetime']]
+
+            # eye location (n by n)
+            self.eye_location = numpy.array([ds.longstore,
```

Comment:
these are already arrays. probably should be
```python
self.eye_location = np.vstack([ds.longstore.values, ds.latstore.values]).T
```

Reply:
Thanks! updated with `numpy.vstack([storm.longstore.values, storm.latstore.values]).T`